packages/core/src/hooks/useContractFunction.ts

```diff
@@ -72,21 +44,21 @@ export function useContractFunction(contract: Contract, functionName: string, op
 
     try {
       transaction = await contractWithSigner[functionName](...args)
-      setState({ transaction, status: 'Mining', chainId })
+      setState({ transaction, status: 'Mining', errored: false, chainId })
```

Comment:
Does errored seem necessary?
I'm thinking that's why errorMessage can be undefined, so users can just check if(errorMessage) //there's an error

Reply:
How about extracting:
```ts
function transactionErrored(transaction) {
  return ...
}
```

Reply:
You mean exposing it to the public as well?

My opinion might not matter, but the reason I see this project as a life saver worth exploring for usage is how it doesn't expose much any unnecessary things, keeping it lightweight, yet adding major features that makes using dapps through hooks viable... I don't know if the goal is different

Reply:
Is errored the new term for failed?

Reply:
Sorry, I misquote your https://github.com/EthWorks/useDApp/pull/202#discussion_r632434360 to extract the errored to function... Seems good now